net/golioth/coap_req.c

```diff
@@ -629,7 +629,12 @@ int golioth_coap_req_sync(struct golioth_client *client,
 	k_sem_take(&sync_data.sem, K_FOREVER);
 
 	if (sync_data.err) {
-		LOG_WRN("req_sync finished with error %d", sync_data.err);
+		if (sync_data.err == -EFAULT) {
+			LOG_WRN("Server reports CoAP Bad Request. (Check payload formatting)");
```

Comment:
Since this is an error reported by the server, it might make more sense to put this warning in the coap reply callback, before it gets converted to EFAULT:

https://github.com/golioth/golioth-zephyr-sdk/blob/main/net/golioth/coap_req.c#L206

Something like:

```c
if (code == COAP_RESPONSE_CODE_BAD_REQUEST) {
    LOG_WRN("Server reports CoAP Bad Request. (Check payload formatting)");
}
```

Reply:
That's a much better place for it. I've made the change, thanks!